File: java/strategy/exercise/exercise2/MessageFormatterContext.java
package strategy.exercise.exercise2;

public class MessageFormatterContext {
    private MessageFormatter messageFormatter;

    public void setFormatter (MessageFormatter messageFormatter) {
        this.messageFormatter = messageFormatter;
    }

    public String formatMessage(String message) {
        return messageFormatter.format(message);
    }

    public String getFormatType() {
        return messageFormatter.getType();
    }
}

File: dart_lang/lib/b_behavioral/b_strategy/c_exercise/exercise_2/message_formatter_context.dart
import 'package:dart_lang/b_behavioral/b_strategy/c_exercise/exercise_2/message_formatter_strategy/message_formatter_strategy.dart';

class MessageFormatterContext{
  late MessageFormatterStrategy _messageFormatter;

  String formatMessage(String message){
    return _messageFormatter.formatMessage(message);
  }

  set formatter(MessageFormatterStrategy strategy) => _messageFormatter = strategy;
  MessageFormatterStrategy get formatType => _messageFormatter;
}